Source/Shaders/ModelExperimental/MaterialStageFS.glsl

```diff
@@ -0,0 +1,217 @@
+czm_modelMaterial defaultModelMaterial()
+{
+    czm_modelMaterial material;
+    material.diffuse = vec3(1.0);
+    material.specular = vec3(0.04); // dielectric (non-metal)
+    material.roughness = 0.0;
+    material.occlusion = 1.0;
+    material.emissive = vec3(0.0);
+    return material;
+}
+
+vec3 SRGBtoLINEAR3(vec3 srgbIn) 
+{
+    return pow(srgbIn, vec3(2.2));
+}
+
+vec4 SRGBtoLINEAR4(vec4 srgbIn) 
+{
+    vec3 linearOut = pow(srgbIn.rgb, vec3(2.2));
+    return vec4(linearOut, srgbIn.a);
+}
+
+#ifdef HAS_NORMALS
+vec3 computeNormal()
+{
+    vec3 ng = normalize(v_normal);
+
+    vec3 normal = ng;
+    #ifdef HAS_NORMAL_TEXTURE
+        vec2 normalTexCoords = TEXCOORD_NORMAL;
+            #ifdef HAS_NORMAL_TEXTURE_TRANSFORM
```

Comment:
Why is this indented further in and the one below not?

Reply:
ah good catch, this one was incorrect.

I really don't know the best way to indent code with `#ifdef`s in a readable way. I've been going with the rule "if I nest an `#ifdef`, indent that block 1 more level"